### PR TITLE
feat(audit): add audit config loader (sibling audit block + lens flags)

### PR DESCRIPTION
**Pull request by howarewoo**
## Affected spec sections

- [ ] `spec/architecture.md`
- [ ] `spec/frameworks.md`
- [ ] `spec/infrastructure.md`
- [ ] `spec/patterns.md`
- [ ] `spec/development.md`
- [ ] `spec/bootstrap.md`
- [ ] `README.md` / other top-level docs

## Checklist

- [ ] Cross-links to related sections still resolve
- [ ] No version numbers hard-coded in `frameworks.md` where "latest" suffices

### CODE REVIEW

**Comment by howarewoo** [comment]
* **#428** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/428?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#427** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/427?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#426** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/426?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#425** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/425?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#424** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/424?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/howarewoo/woostack/424?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#423** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/423?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#422** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/422?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#421** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/421?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: skills/woostack-audit/scripts/tests/test-load-audit-config.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# Sibling review block is ignored, not an error.
+run '{"review":{"severity_floor":"low"},"audit":{"severity_floor":"medium"}}'
+assert_eq "$EC" "0" "sibling review block ignored"
+assert_contains "$(cat "$OUTDIR/config.json")" "medium" "audit severity_floor applied"
+
+# Lens flag simplify keeps the floor, drops production-readiness from force.
+run '{}' 'simplify'; assert_eq "$EC" "0" "lens ok"
+cfg="$(cat "$OUTDIR/config.json")"
+assert_contains "$cfg" "simplify" "lens simplify forces simplify"
+assert_not_contains "$cfg" "production-readiness" "lens simplify drops prod-readiness"
```
Comment (howarewoo) [inline skills/woostack-audit/scripts/tests/test-load-audit-config.sh:31]: **Nit: lens=prod branch has no test**

The simplify lens is exercised (lines 28-31), but the `lens=prod` branch in load-audit-config.sh (lines 30-31: `elif lens == "prod": force = ["production-readiness"]`) has no corresponding test case. That branch drops `simplify` from force and keeps only `production-readiness` — the symmetric counterpart to the simplify-only path. A regression in the `prod` branch (e.g. accidentally leaving both angles in `force`) would go undetected.

Fix: Add a prod lens test after the simplify block: `run '{}' 'prod'`, then assert `production-readiness` is present and `simplify` is absent in `$OUTDIR/config.json`.

<sub>— <strong>MEDIUM · NIT</strong> · flagged by the <code>tests</code> agent</sub>

File: skills/woostack-audit/scripts/tests/test-load-audit-config.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+# Unknown audit key -> loud non-zero.
+run '{"audit":{"bogus":1}}'; assert_eq "$EC" "1" "unknown audit key rejected"
```
Comment (howarewoo) [inline skills/woostack-audit/scripts/tests/test-load-audit-config.sh:58]: **Nit: angles:null config edge case not covered**

load-audit-config.sh line 32 guards against a JSON `null` `angles` value with `ang = audit.get("angles", {}) or {}`. The test suite does not include a case where the config contains `{"audit":{"angles":null}}`. JSON explicitly permits `null` as a value, and a user who writes `angles: null` to reset the field would exercise this branch. Without a test the `or {}` guard is nominally correct but unverified.

Fix: Add a test before `finish`: `run '{"audit":{"angles":null}}'; assert_eq "$EC" "0" "angles null handled"` to confirm the loader exits cleanly and the output config is still valid JSON.

<sub>— <strong>MEDIUM · NIT</strong> · flagged by the <code>tests</code> agent</sub>

**Review by howarewoo — COMMENTED** [review]
## AI Deep Code Review Summary

Round-1 full review of the audit config loader. The implementation is sound: `angles:null` is correctly null-coalesced, JSON parsing is safe, and the test helper exports OUTDIR directly (no command-substitution loss). Two non-blocking test-coverage nits: the `lens=prod` branch and the `angles:null` edge case are not exercised by the test.

---
**Status: APPROVED** — No blocking findings, 2 nits. See inline comments.
*Audited by woostack-review · Host: Claude Code · Provider: anthropic · Model: claude-opus-4-8 (validators), claude-sonnet-4-6 + claude-haiku-4-5 (angles)*



_Review event downgraded to COMMENT — GitHub blocks APPROVE on your own PR. Status line above carries the actual verdict._

File: skills/woostack-audit/scripts/tests/test-load-audit-config.sh (hunk 1 above)
Comment (howarewoo) [reply skills/woostack-audit/scripts/tests/test-load-audit-config.sh:31]: Fixed in `5bac621` — added a `lens=prod` test case asserting the config forces `production-readiness` and drops `simplify`.

File: skills/woostack-audit/scripts/tests/test-load-audit-config.sh (hunk 2 above)
Comment (howarewoo) [reply skills/woostack-audit/scripts/tests/test-load-audit-config.sh:58]: Fixed in `5bac621` — added an `angles:null` test case asserting the `or {}` coalesce guard yields exit 0 (coerced, not an error).

File: skills/woostack-audit/scripts/load-audit-config.sh [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+VALID_KEYS='angles severity_floor ignore models chunking report_dir'
+
+python3 - "$CFG_FILE" "$LENS" "$OUTDIR/config.json" "$VALID_KEYS" <<'PY'
+import json, sys, os
+cfg_file, lens, out, valid_keys = sys.argv[1], sys.argv[2], sys.argv[3], set(sys.argv[4].split())
+audit = {}
+if os.path.exists(cfg_file):
+    with open(cfg_file) as f:
+        try:
+            audit = (json.load(f) or {}).get("audit", {}) or {}
```
Comment (howarewoo) [inline skills/woostack-audit/scripts/load-audit-config.sh:22]: **Nit: Invalid-JSON error path has no test**

Lines 22-23 catch `json.JSONDecodeError` and exit 1 with an `::error` annotation, but no test exercises this branch. The `run` helper writes its first argument verbatim to the config file, so passing malformed JSON (e.g., `run 'not{valid'`) would drive this path. If the handler were accidentally removed or the exit code changed, all existing tests would still pass.

Fix: Add a test case that passes malformed JSON as the config (e.g., `run 'not{valid'`), asserts EC=1, and confirms the script exits non-zero cleanly rather than producing a partial output file.

<sub>— <strong>LOW · NIT</strong> · flagged by the <code>tests</code> agent</sub>

File: skills/woostack-audit/scripts/load-audit-config.sh [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        except json.JSONDecodeError as e:
+            sys.stderr.write("::error file=%s::invalid JSON: %s\n" % (cfg_file, e)); sys.exit(1)
+bad = [k for k in audit if k not in valid_keys]
+if bad:
+    sys.stderr.write("::error file=%s::unknown audit key(s): %s\n" % (cfg_file, ", ".join(bad))); sys.exit(1)
+force = ["simplify", "production-readiness"]
```
Comment (howarewoo) [inline skills/woostack-audit/scripts/load-audit-config.sh:28]: **Nit: Unknown lens silently falls back — behavior unpinned**

Lines 28-31 branch on `lens == "simplify"` and `lens == "prod"` then fall through to the default `force = ["simplify", "production-readiness"]` for any other non-empty value. A caller who passes `AUDIT_LENS=production` (typo) or an unrecognized name gets the full-angle run with no error or warning. No test pins this contract, so neither the silent-fallback behavior nor a hypothetical loud-rejection would be caught by the test suite.

Fix: Add a test case for an unrecognized lens value (e.g., `run '{}' 'bogus-lens'`) and assert either EC=1 with a meaningful error if the intent is to reject unknown lenses, or EC=0 with output matching the no-lens default if silent fallback is intentional. Either way, pin the behavior so a future change cannot silently alter the contract.

<sub>— <strong>MEDIUM · NIT</strong> · flagged by the <code>tests</code> agent</sub>

File: skills/woostack-audit/scripts/tests/test-load-audit-config.sh [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+run '{}' 'prod'; assert_eq "$EC" "0" "lens prod ok"
+cfg="$(cat "$OUTDIR/config.json")"
+assert_contains "$cfg" "production-readiness" "lens prod forces production-readiness"
+assert_not_contains "$cfg" "simplify" "lens prod drops simplify"
+
+# angles:null is coerced to {} by the `or {}` guard, not treated as an error.
+run '{"audit":{"angles":null}}'; assert_eq "$EC" "0" "angles:null coerced, not an error"
+
+# Unknown audit key -> loud non-zero.
+run '{"audit":{"bogus":1}}'; assert_eq "$EC" "1" "unknown audit key rejected"
```
Comment (howarewoo) [inline skills/woostack-audit/scripts/tests/test-load-audit-config.sh:58]: **Nit: User-config angles.force/skip merge path untested**

load-audit-config.sh lines 34-35 merge user-supplied `angles.force` and `angles.skip` from the audit block into the output config (`force + (ang.get("force", []) or [])` / `["architecture"] + (ang.get("skip", []) or [])`). The existing tests exercise only `angles:null` and the absent-angles case (both reduce `ang` to `{}`), so the actual merge when a user provides non-empty force or skip lists is never exercised. A regression in concat order or the `or []` guard would go undetected.

Fix: Add a test case before `finish`: supply `{"audit":{"angles":{"force":["security"],"skip":["react"]}}}`, assert EC=0, then assert the output contains both `security` (user-specified force angle) and `react` (user-specified skip angle) alongside the computed defaults.

<sub>— <strong>MEDIUM · NIT</strong> · flagged by the <code>tests</code> agent</sub>

**Review by howarewoo — COMMENTED** [review]
## AI Deep Code Review Summary

Round-2 re-review. The round-1 lens=prod and angles:null test gaps are fixed. Three further non-blocking test-coverage nits surfaced: the invalid-JSON error path, the unknown-lens fallback contract, and the user-supplied angles.force/skip merge path are not yet exercised. No blocking issues.

---
**Status: APPROVED** — No blocking findings, 3 nits. See inline comments.
*Audited by woostack-review · Host: Claude Code · Provider: anthropic · Model: claude-opus-4-8 (validators), claude-sonnet-4-6 + claude-haiku-4-5 (angles)*



_Review event downgraded to COMMENT — GitHub blocks APPROVE on your own PR. Status line above carries the actual verdict._

File: skills/woostack-audit/scripts/load-audit-config.sh (hunk 3 above)
Comment (howarewoo) [reply skills/woostack-audit/scripts/load-audit-config.sh:22]: Fixed in `85514a4` — added an invalid-JSON test asserting exit 1 (JSONDecodeError path).

File: skills/woostack-audit/scripts/load-audit-config.sh (hunk 4 above)
Comment (howarewoo) [reply skills/woostack-audit/scripts/load-audit-config.sh:28]: Fixed in `85514a4` — added an unknown-lens test asserting exit 0 and both audit angles retained.

File: skills/woostack-audit/scripts/tests/test-load-audit-config.sh (hunk 5 above)
Comment (howarewoo) [reply skills/woostack-audit/scripts/tests/test-load-audit-config.sh:58]: Fixed in `85514a4` — added a user angles.force/skip test asserting they append onto the audit defaults.

File: skills/woostack-audit/scripts/tests/test-load-audit-config.sh [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+run '{}' 'prod'; assert_eq "$EC" "0" "lens prod ok"
+cfg="$(cat "$OUTDIR/config.json")"
+assert_contains "$cfg" "production-readiness" "lens prod forces production-readiness"
+assert_not_contains "$cfg" "simplify" "lens prod drops simplify"
+
+# angles:null is coerced to {} by the `or {}` guard, not treated as an error.
+run '{"audit":{"angles":null}}'; assert_eq "$EC" "0" "angles:null coerced, not an error"
+
+# User-supplied angles.force/skip merge onto the audit defaults (append, not replace).
+run '{"audit":{"angles":{"force":["bugs"],"skip":["docs"]}}}'; assert_eq "$EC" "0" "user angles merge ok"
```
Comment (howarewoo) [inline skills/woostack-audit/scripts/tests/test-load-audit-config.sh:43]: **Nit: Merge test omits assertion that defaults survive**

The 'user angles merge' test verifies that user-supplied force values ("bugs") and skip values ("docs") appear in the output, but never asserts that the audit defaults ("simplify", "production-readiness") are still present in the force list after the merge. If the implementation changed `force + ang.get("force", [])` to just `ang.get("force", [])`, the user-supplied values would still pass but the defaults would silently disappear, and this test would not catch the regression.

Fix: After the two existing assert_contains calls on lines 45-46, add assertions that the defaults are preserved: assert_contains "$cfg" "simplify" "defaults preserved after user merge" and assert_contains "$cfg" "production-readiness" "defaults preserved after user merge".

<sub>— <strong>LOW · NIT</strong> · flagged by the <code>bugs</code> agent</sub>

**Review by howarewoo — COMMENTED** [review]
## AI Deep Code Review Summary

Round-3 (final) review. The round-2 test gaps (invalid-JSON, unknown-lens fallback, user angles merge) are all covered; the test now has 21 assertions and the tests angle is satisfied. One residual non-blocking nit: the merge test asserts the user-supplied values appear but not that the audit defaults survive the append. Recorded at the review-round cap; the loader code itself correctly appends. No blocking issues.

---
**Status: APPROVED** — No blocking findings, 1 nit. See inline comments.
*Audited by woostack-review · Host: Claude Code · Provider: anthropic · Model: claude-opus-4-8 (validators), claude-sonnet-4-6 + claude-haiku-4-5 (angles)*



_Review event downgraded to COMMENT — GitHub blocks APPROVE on your own PR. Status line above carries the actual verdict._

File: skills/woostack-audit/scripts/tests/test-load-audit-config.sh (hunk 6 above)
Comment (howarewoo) [reply skills/woostack-audit/scripts/tests/test-load-audit-config.sh:43]: Accepted nit at the review-round cap (max_rounds=3), left open as the record. The merge logic in `load-audit-config.sh` (`force + (ang.get("force", []) or [])`) provably appends, so the defaults are present; adding an assertion that `simplify`/`production-readiness` survive the merge is a worthwhile one-line follow-up but non-blocking. No blocking findings remain.

**Comment by howarewoo** [comment]
### Merge activity

* **Jun 26, 1:59 PM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/howarewoo/woostack/424).
* **Jun 26, 2:04 PM UTC**: [Graphite](https://app.graphite.com/github/pr/howarewoo/woostack/424) rebased this pull request as part of a merge.
* **Jun 26, 2:04 PM UTC**: @howarewoo merged this pull request with [Graphite](https://app.graphite.com/github/pr/howarewoo/woostack/424).
